fixing a small bug in multisegment well

GitPaean opened the pull request:
which is due to the previous report step to be zero-rate control, we might not get exactly 0 value rates. Doing division with it is causing problem.

GitPaean committed:
fixing a small bug in multisegment well

which is due to the previous report step to be zero-rate control, we
might not get exactly 0 value rates. Doing division with it is causing
problem.

GitPaean commented:
jenkins build this please

GitPaean commented:
jenkins build this please

atgeirr commented on opm/autodiff/MultisegmentWell_impl.hpp:
```diff
@@ -339,8 +339,9 @@ namespace Opm
 
-                if (original_rates_under_phase_control != 0.0 ) {
-                    double scaling_factor = target / original_rates_under_phase_control;
+                // sometimes the previous report step is zero-rate control, it causes this value to be some extremely small value
```
Two issues:
 - Is original_rates_under_phase_control always positive? I think it can be negative, and that must be taken into account.
 - The 1.0e-10 should be a constant with a name, such as rate_threshold.

GitPaean committed:
same small fix to StandardWell

which is due to the previous report step to be zero-rate control, we
might not get exactly 0 value rates. Doing division with it might cause
problems.

GitPaean commented:
jenkins build this please

atgeirr commented:
jenkins build this please

GitPaean commented:
This PR can wait, I think.

GitPaean commented:
Maybe we need this PR to fix the new problem of NaN. I really do not understand why it happens sometimes, not more often.

GitPaean commented:
The current NaN happens, when a well used to have zero-rate water control, then there is a new non-zero rate GAS control.

bska commented:
@GitPaean : Is this PR still relevant?  If so, what is the cost of rebasing it onto current master versus starting from scratch?

bska commented:
@GitPaean : Is this work still relevant?

GitPaean commented:
This part of the code has evolved much and I think we can close the PR.
